src/main/java/com/spotify/folsom/client/DefaultRawMemcacheClient.java

```diff
@@ -259,6 +270,10 @@ public void write(final ChannelHandlerContext ctx, final Object msg,
                       final ChannelPromise promise)
         throws Exception {
       Request<?> request = (Request<?>) msg;
+      if (request.isCancelled()) {
```

Comment:
Yes, good point.

Though I think it's perhaps best to simply abandon this change, I don't think the added complexity can be justified.